src/rp2040/rp2040_link.lds.S

```diff
@@ -0,0 +1,66 @@
+// rp2040 linker script (based on armcm_link.lds.S and customized for stage2)
+//
+// Copyright (C) 2019-2021  Kevin O'Connor <kevin@koconnor.net>
+//
+// This file may be distributed under the terms of the GNU GPLv3 license.
+
+#include "autoconf.h" // CONFIG_FLASH_START
+
+OUTPUT_FORMAT("elf32-littlearm", "elf32-littlearm", "elf32-littlearm")
+OUTPUT_ARCH(arm)
+
+MEMORY
+{
+  rom (rx) : ORIGIN = CONFIG_FLASH_START , LENGTH = CONFIG_FLASH_SIZE
+  ram (rwx) : ORIGIN = CONFIG_RAM_START , LENGTH = CONFIG_RAM_SIZE
+}
+
+SECTIONS
+{
+    .text : {
+        . = ALIGN(4);
+        KEEP(*(.boot2))
+        _text_vectortable_start = .;
+        KEEP(*(.vector_table))
+        _text_vectortable_end = .;
+        *(.text .text.*)
+        *(.rodata .rodata*)
+    } > rom
+
+    . = ALIGN(4);
+    _data_flash = .;
+
+    .data : AT (_data_flash)
+    {
+        . = ALIGN(4);
+        _data_start = .;
+        *(.ramfunc .ramfunc.*);
+        *(.data .data.*);
+        . = ALIGN(4);
+        _data_end = .;
+    } > ram
+
+    .bss (NOLOAD) :
+    {
+        . = ALIGN(4);
+        _bss_start = .;
+        *(.bss .bss.*)
+        *(COMMON)
+        . = ALIGN(4);
+        _bss_end = .;
+    } > ram
+
+    _stack_start = CONFIG_RAM_START + CONFIG_RAM_SIZE - CONFIG_STACK_SIZE - 1024;
```

Comment:
Magical "-1024" is really magical. _stack_start is used to place request key what should be preserved after reset. But rp2040 bootrom  uses and of the ram to copy boot2(256 bytes) from flash and boot2 uses some stack on top of what. so ~300 bytes around CONFIG_RAM_START + CONFIG_RAM_SIZE are damaged by the boot process of the rp2040. That is why we shift stack by 1024 bytes.